BUILDER_DEV.md

```diff
@@ -1,221 +1,67 @@
-* Set up some environmental variables (I put these in my .bashrc)
-
-```
-export HAB_AUTH_TOKEN="<your github token>"
-export HAB_BLDR_URL="http://localhost:9636"
-export HAB_ORIGIN="core"
-```
-
-* symlink /src to path/to/your/habitat/repo
-
-```
-$ sudo ln -s path/to/your/habitat/repo /src
-```
+## Pre-Requisites
 
-* Cd into your Habitat repo
+### GitHub OAuth Application
 
-```
-$ cd /src
-```
+1. Create a GitHub Organization
+1. [Setup a GitHub application](https://github.com/settings/apps/new) for your GitHub organization.
+1. Set the value of `Homepage URL` to `http://${APP_HOSTNAME}`
+1. Set the value of `User authorization callback URL` to `http://${APP_HOSTNAME}/` (The trailing `/` is *important*)
+1. Set the value of `Webhook URL` to `http://${APP_HOSTNAME}/`
+1. Set everything to read only (this is only used for your org so it's safe)
+1. Save and download the pem key
+1. Copy the pem key to `${HABITAT_SRC_ROOT}/.secrets/builder-dev-app.pem`
+1. Record the the client-id, client-secret and app_id. These will be used for the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET` and `GITHUP_APP_ID` build variables (seen below).
 
-* Run the provision script
-
-```
-$ ./support/linux/provision.sh
-```
+### Setup GitHub Teams
 
-* Source cargo
+1. Create a GitHub team in your organization for administrators and retrieve the team-id. This will be used for the value of `GITHUB_ADMIN_TEAM`
 
-```
-$ source $HOME/.cargo/env
-```
+> note: The only way to retrieve the ID for a GitHub team is currently [by the API](https://github.com/rei/github-api-utils/blob/master/bash/get-team-id.sh)
 
-* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
+### Studio requirements
 
-```
-$ make build-srv
-```
+1. Ensure you have run `hab setup` in your environment that will be executing the studio
 
-* Run this command (it may take awhile) to start all the Builder services
+### Starting the services
 
-```
-$ sudo -E make bldr-run-no-build
-```
+From either your VM or Mac:
 
-* Wait for all services to start - they are started when you see worker heartbeat debug entries in the log. If services do not start because some were already running from a previous attempt, run `$ make bldr-kill`
+* `cd <habitat source path>`
+* `direnv allow`
+* `hab studio enter`
+* `start-builder`
 
 ### UI Setup
 
+You should only need to do this if you are developing the UI
+
 * Follow the instructions in the [Web UI README](https://github.com/habitat-sh/habitat/blob/master/components/builder-web/README.md) to get the Web UI running locally.
 * Open up a browser and navigate to http://localhost:3000/#/pkgs - you should see the Habitat UI running locally.
```

Comment:
Should this be `http://${APP_HOSTNAME}/#/pkgs`? Or perhaps `…#/sign-in`?

Reply:
Actually in this case it will be different. APP_HOSTNAME will be the hab package running inside the studio. Currently all UI dev is done on your local host so it'll always run on `localhost:3000`